src/util/misc/projectStroke.ts

```diff
@@ -0,0 +1,93 @@
+import { Point } from "../../point.class";
+import { createVector, getBisector } from "./vectors";
+import { StrokeLineJoin } from '../../typedefs';
+/**
+ * Project stroke width on points returning 2 projections for each point as follows:
+ * - `miter`: 2 points corresponding to the outer boundary and the inner boundary of stroke.
+ * - `bevel`: 2 points corresponding to the bevel boundaries, tangent to the bisector.
+ * - `round`: same as `bevel`
+ * Used to calculate object's bounding box
+ * @static
+ * @memberOf fabric.util
+ * @param {Point[]} points
+ * @param {Object} options
+ * @param {number} options.strokeWidth
+ * @param {'miter'|'bevel'|'round'} options.strokeLineJoin
+ * @param {number} options.strokeMiterLimit https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit
+ * @param {boolean} options.strokeUniform
+ * @param {number} options.scaleX
+ * @param {number} options.scaleY
+ * @param {boolean} [openPath] whether the shape is open or not, affects the calculations of the first and last points
+ * @returns {Point[]} array of size 2n/4n of all suspected points
+ */
+
+type projectStrokeOnPointsOptions = {
+  strokeWidth: number;
+  strokeLineJoin: StrokeLineJoin;
+  strokeMiterLimit: number; // https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit
```

Comment:
if the comment is `/** **/` on top of the prop then it is live and not static text

Reply:
and perhaps it should be a `Pick` type from `TStrokeOptions`

Reply:
i ll recheck this, wasn't really a deep though.